### DISCUSSION TITLE
Visual studio code is not connecting to GitHub Copilot

### DISCUSSION THREAD

**Discussion by AntoineT95** [community/community#43206] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Hello, 

Visual studio code is note connecting to GitHub and especially GitHub Copilot. 

When i want to connect i have this message below:

[](url)
![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/19637540/210374031-37cb7f90-3fb1-4d69-a755-f41c6d652db5.png)

Then i log on my browser:

![MicrosoftTeams-image (3)](https://user-images.githubusercontent.com/19637540/210374193-33c814de-511c-4a63-a422-9836a2a6f91f.png)

And the I have this message error: "GitHub Copilot could not connect to server. Extension activation failed: Cancelled" or 

![MicrosoftTeams-image (4)](https://user-images.githubusercontent.com/19637540/210374377-5043ea7b-202b-4a58-b52d-75ac863356d3.png)

I don't know what to do I have desactived my VPN and the issue is the same. 

Thank you in advance

**Comment by Akash1134** [comment]
Hi @AntoineT95 , please look at the steps [mentioned here](https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-github-copilot-in-visual-studio-code#viewing-network-connectivity-diagnostics-logs) to troubleshoot this issue.

**Reply by AntoineT95** [reply]
Hi, Thank you for the help ! 

This what I have: 

![image](https://user-images.githubusercontent.com/19637540/210393606-dff68e8e-5cc6-438a-be14-1d381a3b9a36.png)

**Comment by haseeb-xd** [comment]
Hello, I'm sorry you're experiencing problems. We've discovered that if Copilot is unable to connect to our servers, the Copilot icon could not appear. Are you protected by a firewall, VPN, or any piece of software that is changing your network connections?

**Reply by AntoineT95** [reply]
Hello, Yes I have firewall and a VPN on my company laptoop. What step should i do to make Github Copilot working with it ?

**Reply by haseeb-xd** [reply]
> Hello, Yes I have firewall and a VPN on my company laptoop. What step should i do to make Github Copilot working with it ?

It seems you can not use Copilot if you are behind VPN or proxies.

**Reply by AntoineT95** [reply]
There is no way to bypass it?

**Reply by haseeb-xd** [reply]
Have you tried disabling and enabling the Github Copilot extension?

**Reply by AntoineT95** [reply]
Yes several times

**Reply by haseeb-xd** [reply]
many people are having issues when they are behind VPN, Firewall or proxies. don't think Github has updated on this recently

**Reply by AntoineT95** [reply]
Hope they will soon !

**Comment by mishra-prajjawal** [comment]
Hi 👋 @AntoineT95  thanks for reaching out I believe that this problem must be occuring due to a VPN network in your device that might be dwindling with the authentication of GitHub.

Lastly check if your subscription is valid because it is possible that it may had been expired.

**Reply by AntoineT95** [reply]
Hello, Yes my subscription is available. Even without the VPN it's not working.

**Comment by Akash1134** [comment]
Hi @AntoineT95, Thank you for your patience and continued efforts in helping resolve the problem. 🙂  As mentioned in the [doc](https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-github-copilot-in-visual-studio-code#viewing-network-connectivity-diagnostics-logs), the issue is linked to a service intercepting your network connecting to our servers. Here's a [similar discussion](https://github.com/community/community/discussions/11016) that might help you troubleshoot further. Let us know how it goes.

**Comment by antonioacg** [comment]
https://marketplace.visualstudio.com/items?itemName=ukoloff.win-ca
